osgar/logger.py

```diff
@@ -58,18 +58,38 @@ def format_header(start_time):
 
 def format_packet(stream_id, data, dt):
     ret = []
-    assert dt.days == 0, dt  # multiple days not supported yet
-    time_frac = (dt.seconds * 1000000 + dt.microseconds) & TIMESTAMP_MASK
+    dt_bytes = format_timedelta(dt)
     index = 0
     while index + 0xFFFF <= len(data):
-        ret.append(struct.pack('IHH', time_frac, stream_id, 0xFFFF))  # header
+        ret.append(dt_bytes + struct.pack('HH', stream_id, 0xFFFF))  # header
         ret.append(data[index:index + 0xFFFF]) # partial data
         index += 0xFFFF
-    ret.append(struct.pack('IHH', time_frac, stream_id, len(data) - index))  # header
+    ret.append(dt_bytes + struct.pack('HH', stream_id, len(data) - index))  # header
     ret.append(data[index:])
     return ret
 
 
+def format_timedelta(dt):
+    assert dt.days == 0, dt  # multiple days not supported yet
+    time_frac = (dt.seconds * 1000000 + dt.microseconds) & TIMESTAMP_MASK
+    return time_frac.to_bytes(4, 'little')
+
+
+def timedelta_parser(start_time=datetime.timedelta()):
```

Comment:
I would add here some description what this function does and why it returns another (internal) function.

Reply:
Let me sketch the description here:

> create timedelta parser
>
> Created parser converts `uint32` microseconds to `timedelta` object. The parser internally tracks overflows in `us_offset`, `prev_microseconds`. The intended use is to sequentially parse incoming timestamps. The argument `start_time` supports creation  of a parser where the last processed timestamp corresponds to the given timedelta object.

Would that do?